--- orig/asmcomp/arm64/emit.mlp	2016-11-04 12:08:24.000000000 -0400
+++ patch/asmcomp/arm64/emit.mlp	2018-11-14 23:47:13.000000000 -0500
@@ -969,6 +969,7 @@
   `	.globl	{emit_symbol lbl_end}\n`;
   `{emit_symbol lbl_end}:\n`;
   `	.long	0\n`;
+  `	.long	0\n`;			(* For quad alignment of frametable *)
   let lbl = Compilenv.make_symbol (Some "frametable") in
   `	.globl	{emit_symbol lbl}\n`;
   `{emit_symbol lbl}:\n`;
--- orig/asmcomp/amd64/emit.mlp	2016-11-04 12:08:24.000000000 -0400
+++ patch/asmcomp/amd64/emit.mlp	2018-11-14 23:47:18.000000000 -0500
@@ -1074,6 +1074,7 @@
   D.data ();
   emit_global_label "data_end";
   D.long (const 0);
+  D.long (const 0);			(* For quad aligment of frametable *)
 
   emit_global_label "frametable";
 
